### DISCUSSION TITLE
Arbitrary laser profile?

### DISCUSSION THREAD

**Discussion by bzdjordje** [epochpic/epoch#173] [Q&A]
Is it possible to initialize the laser with an arbitrary profile from a secondary input file? I am looking to include higher-order spectral phase into my models and I see that there is an interpolate function. If I wanted to model an arbitrary laser profile do I have to include a full array of relevant points? Can the phase variable also be described by a function/input array and not just a single scalar?

**Comment by TomGoffrey** [comment]
There's no option to specify the laser via an external file. However, you can specify the laser phase (and several other parameters) as a functions of time and space.

**Reply by bzdjordje** [reply]
Would giving a time-dependent phase profile be sufficient to model dazzler effects and give a laser profile such as the following?

![image](https://user-images.githubusercontent.com/4984500/111401509-a6604a00-8686-11eb-9231-3ebeb64e818e.png)
